### PR TITLE
Add Default Redirect Path for Auth Requests From Different Origin

### PR DESCRIPTION
**Pull request by mendelB**
### What does this PR do?

This PR ensures that authentication requests to `/next/login` originating from URL's of a different domain will be redirected post-auth to the default `$this->redirectTo` path and not the previous page from whence they came.

### Any background context you want to provide?
Over at Northstar, password reset form submissions are redirected to `/next/login` to authenticate but are then abruptly returned to the reset password form. Cases such as those (where the previous URL is not of the same domain as the Phoenix application) will be better redirected to a sensible Phoenix location (currently set to `/campaigns`).

### What are the relevant tickets/cards?

Refs [Pivotal ID #164504959](https://www.pivotaltracker.com/story/show/164504959)

### CODE REVIEW

File: app/helpers.php [hunk 1]
```diff
@@ -443,3 +443,17 @@ function generate_streamed_csv($columns, $records)
...
     return fclose($file);
 }
+
+/**
+ * Determine if URL is of the same origin as the app's URL.
+ *
+ * @param  string $url
+ * @return bool
+ */
+function is_same_origin($url)
```
Comment (mendelB) [inline app/helpers.php]: `is_same_domain`? `is_same_host`? Ah, naming... 😩

File: app/Http/Controllers/AuthController.php [hunk 2]
```diff
@@ -42,11 +42,15 @@ public function getLogin(ServerRequestInterface $request, ResponseInterface $res
     {
         $queryParams = $request->getQueryParams();
 
-        // Save the post-login redirect for when the user completes the flow: either to the intended
-        // page (if logging in to view a page protected by the 'auth' middleware), or the previous
-        // page (if the user clicked "Log In" in the top navigation).
+        // Save the post-login redirect for when the user completes the flow.
```
Comment (mendelB) [inline app/Http/Controllers/AuthController.php]: Rearranging these comments to have it make sense with the new flow was tricky... Is it any clearer now?

File: app/helpers.php (hunk 1 above)
Comment (weerd) [reply app/helpers.php]: maybe `is_same_domain` is more straightforward? 🤔

Comment (mendelB) [reply app/helpers.php]: Yeah I think so! I was about to update that I just saw a similar helper in Northstar with `domain`
https://github.com/DoSomething/northstar/blob/84d387a9d83b5ed6e7bcf05899bd05d687b6808a/app/helpers.php#L203

Comment (mendelB) [reply app/helpers.php]: I'm going to add the safe hard return that method has as well.

**Comment by aaronschachter** [comment]
Just checking -- do these changes need corresponding updates in [Northstar](https://github.com/DoSomething/northstar/blob/84d387a9d83b5ed6e7bcf05899bd05d687b6808a/app/Http/Controllers/Web/ResetPasswordController.php#L109)?

**Comment by mendelB** [comment]
@aaronschachter this shouldn't require any updates in Northstar!

**Pull request merged**